src/jdk.compiler/share/classes/com/sun/source/tree/PatternTree.java

```diff
@@ -26,17 +26,9 @@
 package com.sun.source.tree;
 
 /**
- * {@preview Associated with pattern matching for instanceof, a preview feature of
- *           the Java language.
- *
- *           This interface is associated with <i>pattern matching for instanceof</i>, a preview
- *           feature of the Java language. Preview features
- *           may be removed in a future release, or upgraded to permanent
- *           features of the Java language.}
- *
  * A tree node used as the base class for the different kinds of
  * statements.
  *
- * @since 14
+ * @since 16
  */
 public interface PatternTree extends Tree {}
```

Comment:
I think that this interface is there for forward compatibility, it is still weird to have an empty interface and then an empty class in JCTree implementing this interface. Doesn't it make sense to move any method from BindingPatternTree up to this interface? I think that if having this empty interface is deemed unavoidable then, just an idea, we should acknowledge the fact that this is a forward looking interface.

Reply:
We have multiple such interfaces - like StatementTree and ExpressionTree. It is a forward-looking interface, but I think necessary: InstanceofTree.getPattern() needs to return something, and having a PatternTree is (I think) better than returning generic Tree, or BindingPatternTree (the latter is likely to be insufficient sometime soon).